### PR TITLE
Prefix operator names

### LINKED ISSUE #311: Pattern matching with prefix form of operators is broken
```haskell
(+) x y = x
```
is transformed into
```haskell
+ x y = x
```

### PR DESCRIPTION
**Pull request by adamse**
Subsumes #319, #320. Fixes #311, #308.

### CODE REVIEW

**Comment by chrisdone** [comment]
I've been away and I'm a bit confused about what this PR is for now that it conflicts with others. Can you summarize? The commits contain merges and commits from other people.

**Comment by chrisdone** [comment]
Oops, hit `Close and comment` instead of `Comment`.

**Comment by adamse** [comment]
This just makes a function out of the fixes for the mentioned issues instead of inlining the code 3-4 times. The other persons commit is the test case: I used his as a courtesy.

**Comment by chrisdone** [comment]
Can't tell what I'm merging here.
